Marlin/src/lcd/extui/lib/mks_ui/draw_more.cpp

```diff
@@ -0,0 +1,267 @@
+//#include "../lvgl/src/lv_objx/lv_img.h"
+//#include "../lvgl/src/lv_core/lv_disp.h"
+//#include "../lvgl/src/lv_core/lv_refr.h"
+#include "draw_ui.h"
+#include "../../../../gcode/queue.h"
+
+extern lv_group_t * g;
+static lv_obj_t * scr;
+
+#define ID_CUSTOM_1   1
+#define ID_CUSTOM_2   2
+#define ID_CUSTOM_3   3
+#define ID_CUSTOM_4   4
+#define ID_CUSTOM_5   5
+#define ID_CUSTOM_6   6
+#define ID_CUSTOM_7   7
+#define ID_M_RETURN   8
+
+static void event_handler(lv_obj_t * obj, lv_event_t event) {
+  switch (obj->mks_obj_id) {
+    case ID_CUSTOM_1:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_1_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_1));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_2:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_2_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_2));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_3:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_3_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_3));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_4:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_4_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_4));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_5:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_5_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_5));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_6:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_6_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_6));
+        #endif
+      }
+      break;
+    case ID_CUSTOM_7:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        #if ENABLED(USER_CMD_7_ENABLE)
+          queue.inject_P(PSTR(USER_GCODE_7));
+        #endif
+      }
+      break;
+    case ID_M_RETURN:
+      if (event == LV_EVENT_CLICKED) {
+      }
+      else if (event == LV_EVENT_RELEASED) {
+        lv_clear_more();
+        lv_draw_tool();
+      }
+      break;
+  }
+}
+
+void lv_draw_more(void) {
+  #if ENABLED(USER_CMD_1_ENABLE)
+    lv_obj_t *buttonCustom1;
+  #endif
+  #if ENABLED(USER_CMD_2_ENABLE)
+    lv_obj_t *buttonCustom2;
+  #endif
+  #if ENABLED(USER_CMD_3_ENABLE)
+    lv_obj_t *buttonCustom3;
+  #endif
+  #if ENABLED(USER_CMD_4_ENABLE)
+    lv_obj_t *buttonCustom4;
+  #endif
+  #if ENABLED(USER_CMD_5_ENABLE)
+    lv_obj_t *buttonCustom5;
+  #endif
+  #if ENABLED(USER_CMD_6_ENABLE)
+    lv_obj_t *buttonCustom6;
+  #endif
+  #if ENABLED(USER_CMD_7_ENABLE)
+    lv_obj_t *buttonCustom7;
+  #endif
```

Comment:
These should all be combined with their initialization. This will save some lines of code and make the file cleaner.